process/factory/shard/vmContainerFactory.go

```diff
@@ -89,9 +137,24 @@ func (vmf *vmContainerFactory) Create() (process.VirtualMachinesContainer, error
 		return nil, err
 	}
 
+	// The vmContainerFactory keeps a reference to the container it has created,
+	// in order to replace, from within the container, the VM instances that
+	// become out-of-date after specific epochs.
+	vmf.container = container
+	vmf.epochNotifier.RegisterNotifyHandler(vmf)
+
 	return container, nil
 }
 
+// EpochConfirmed updates VMs in all created containers depending on the epoch
+func (vmf *vmContainerFactory) EpochConfirmed(epoch uint32) {
+	if vmf.config.OutOfProcessEnabled {
```

Comment:
outofprocess should be per epoch.

Reply:
Created flag per epoch called `OutOfProcessSupported` for each version.